@3rdweb-sdk/react/hooks/useCommon.ts

```diff
@@ -375,3 +380,56 @@ export function useAirdropMutation<TContract extends ValidContractInstance>(
     },
   );
 }
+
+export function useBurnMutation<TContract extends ValidContractInstance>(
+  contract?: TContract,
+) {
+  const contractType = useContractTypeOfContract(contract);
+
+  return useMutationWithInvalidate(
+    async (burnData: IBurnInput) => {
+      invariant(
+        contract,
+        "Contract is not a valid contract. Please use a valid contract",
+      );
+      invariant(
+        "burn" in contract,
+        "Contract does not support burn functionality",
+      );
+      if (contract instanceof Edition || contract instanceof EditionDrop) {
+        return await contract.burn(burnData.tokenId, burnData.amount || 1);
+      } else if (
+        contract instanceof NFTCollection ||
+        contract instanceof NFTDrop
+      ) {
+        return await contract.burn(burnData.tokenId);
+      } else if (contract instanceof Token) {
+        return await contract.burn(burnData.amount || 0);
+      }
+      throw new Error("Contract is not a valid contract");
+    },
+    {
+      onSuccess: (_data, _variables, _options, invalidate) => {
+        // this should not be possible, but we need to catch it in case it does
+        // if we don't know we just invalidate everything.
+        if (!contractType) {
+          return invalidate(
+            Object.keys(CacheKeyMap)
+              .map((key) => {
+                const cacheKeys = CacheKeyMap[key as keyof typeof CacheKeyMap];
+                if ("list" in cacheKeys) {
+                  return cacheKeys.list(contract?.getAddress());
+                }
+                return undefined as never;
+              })
+              .filter((fn) => !!fn),
+          );
+        }
+
+        return invalidate([
+          CacheKeyMap[contractType].list(contract?.getAddress()),
```

Comment:
this needs to also invalidate the `supply` query ("supply" goes down when burning)